adi/adxl345.py

```diff
@@ -0,0 +1,101 @@
+#       on or directly connected to an Analog Devices Inc. component.
+#
+# THIS SOFTWARE IS PROVIDED BY ANALOG DEVICES "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
+# INCLUDING, BUT NOT LIMITED TO, NON-INFRINGEMENT, MERCHANTABILITY AND FITNESS FOR A
+# PARTICULAR PURPOSE ARE DISCLAIMED.
+#
+# IN NO EVENT SHALL ANALOG DEVICES BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, INTELLECTUAL PROPERTY
+# RIGHTS, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR
+# BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+# STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF
+# THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+from decimal import Decimal
+
+import numpy as np
+from adi.attribute import attribute
+from adi.context_manager import context_manager
+
+
+class adxl345(context_manager, attribute):
+    """ ADXL345 3-axis accelerometer """
+
+    channel = [3]
+    _device_name = "adxl345"
+    _rx_data_type = np.int32
+
+    def __init__(self, uri=""):
+
+        context_manager.__init__(self, uri, self._device_name)
+        self._ctrl = self._ctx.find_device("adxl345")
+        self.accel_x = self._channel(self._ctrl, "accel_x")
+        self.accel_y = self._channel(self._ctrl, "accel_y")
+        self.accel_z = self._channel(self._ctrl, "accel_z")
+
+    @property
+    def sampling_frequency_available(self):
+        """Provides all available sampling frequency settings for the ADXL345 channels"""
+        return self._get_iio_dev_attr("sampling_frequency_available")
+
+    class _channel(attribute):
+        """ADXL345 channel"""
+
+        def __init__(self, ctrl, channel_name):
+            self.name = channel_name
+            self._ctrl = ctrl
+
+        @property
+        def calibbias(self):
+            """ADXL345 channel offset"""
+            return self._get_iio_attr(self.name, "calibbias", False)
+
+        @calibbias.setter
+        def calibbias(self, value):
+            self._set_iio_attr(self.name, "calibbias", False, value)
+
+        @property
+        def raw(self):
+            """ADXL345 channel raw value"""
+            return self._get_iio_attr(self.name, "raw", False)
+
+        @property
+        def sampling_frequency(self):
+            """ADXL345 channel sampling frequency"""
+            return float(self._get_iio_attr_str(self.name, "sampling_frequency", False))
+
+        @sampling_frequency.setter
+        def sampling_frequency(self, value):
+            self._set_iio_attr(
+                self.name, "sampling_frequency", False, str(Decimal(value).real)
+            )
+
+        @property
+        def scale(self):
+            """ADXL345 channel scale(gain)"""
+            return float(self._get_iio_attr_str(self.name, "scale", False))
+
+    def to_g(self, ch):
```

Comment:
This functionality might already exist: https://github.com/analogdevicesinc/pyadi-iio/blob/master/adi/rx_tx.py#L148
aka setting the rx_output_type property will get you the same thing

Reply:
Looks like this is for buffered channels only? It might be handy to have a general-purpose (unbuffered) channel class that includes SI conversion.

Reply:
Sounds reasonable. Have an idea of what the API would look like? Maybe:
`data = dev.rx_unbuffered()`
or 
`data = dev.rx('unbuffered')`
Just spitballing here